### DISCUSSION TITLE
Almalinux install times out because kickstart file is never downloaded.

### DISCUSSION THREAD

**Discussion by akulbe** [vmware/packer-examples-for-vsphere#447] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

main

### VMware vSphere

8.0.0

### HashiCorp Packer

1.8.5

### HashiCorp Packer Plugin for VMware vSphere

1.1.1

### Guest Operating System

AlmaLinux 9

### Environment Details

vSphere & vCenter 8.0

running the packer stuff repo from a VM on this host

### Description

Anaconda is failing to fetch the kickstart file, even though the HTTP server is starting.

### Expected Behavior

Anaconda would get the kickstart file, AlmaLinux install would complete successfully as expected.

### Actual Behavior

The KS file is never downloaded, and the installation times out.
![almalinux_install_timeout](https://user-images.githubusercontent.com/123279/210573389-cb1afc47-589f-40ea-89f5-28773d4cc161.jpg)

### Steps to Reproduce

Run set-envvars.sh, supply the following answers:

```
[akulbe@workspace packer-examples-for-vsphere]$ ./set-envvars.sh

> Set the vSphere credentials.
Enter the FQDN of your vCenter Server instance: vcsa.kulbe.cloud
Enter the username for the user account: administrator@vsphere.local
Enter the password for the user account:
Skip vCenter Server instance SSL verification? (y/n): y

> Set the vSphere settings.
Enter the vSphere datacenter name: kulbe-dc
Enter the vSphere cluster name: kulbe-cluster
Enter the datastore name virtual machines: datastore2
Enter the network name: VM Network
Enter the folder name: Packer Templates
Enter the content library name: LVB
Enter the datastore name for .iso files: datastore1

> Set the common virtual machine settings.
Enter the virtual hardware version (recommended: 19): 20
Enable VMware Tools upgrade (recommended: true): true
Remove CD-ROMs (recommended: true): true

> Set the common template and content library settings.
Convert to a template (recommended: false): false
Export OVF Template to the content library (recommended: true): true
Destroy virtual machine artifact (recommended: true): true
Skip export to content library (recommended: false): false

> Set the common OVF export settings.
Export as an OVF artifact (recommended: false): false
Overwrite an existing OVF artifact (recommended: true): true

> Set the common boot and provisioning settings.
Set the provisioning datasource (recommended: http): http
Enter the IP address of the interface on this host: 172.20.252.250
Enter the starting HTTP port (recommended: 8000): 8000
Enter the ending HTTP port (recommended: 8099): 8099
Enter the IP wait timeout (recommended: 20m): 20m
Enter the virtual machine shutdown timeout (recommended: 15m): 15m

> Set proxy credentials.
Use SOCKS Proxy? (y/n): n

> Set the default user account credentials.
Enter the username for the account: aaron
Enter the password for the account:
Enter the SHA-512 encrypted password for the account:
Enter the key for the account:

> Set the Ansible credentials.
Enter the username for the account: ansible
Enter the key for the account:

> Set the RedHat Subscription Manager credentials.
Enter the username for the account: longviewbits
Enter the password for the account:

> Set the SUSE Customer Center credentials.
Enter the email for the account:
Enter the code for the account:

> Set the HCP Packer registry.
Enable the HCP Packer registry: no


> Setting the vSphere credentials...
> Setting the vSphere settings...
> Setting the common virtual machine settings...
> Setting the common template and content library settings...
> Setting the OVF export settings...
> Setting the common boot and provisioning settings...
> Setting the default account credentials...
> Setting the Ansible credentials...
> Setting the RedHat Subscription Manager credentials...

> Setting the SUSE Customer Center credentials...

> Setting the HCP Packer...

Display the environment variables? (y/n): n
[akulbe@workspace packer-examples-for-vsphere]$
```

### Screenshots

![almalinux_build_output](https://user-images.githubusercontent.com/123279/210574052-197dc8e7-78fa-409a-b1f3-9b47b9d0da68.jpg)
![almalinux_install_timeout](https://user-images.githubusercontent.com/123279/210574070-b889e5a9-dd06-4eab-822c-849f03e0bd19.jpg)

### Additional Context

I realize this could be a configuration issue on my part, but I'm honestly not sure.

**Comment by tenthirtyam** [comment, marked as answer]
I’ve converted the issue to a discussion - all builds were tested in `main` yesterday. 

This appears to be a configuration issue with the machine image being build not having access back to the Packer host’s HTTP server - you can can switch to disk-based mode per the README to test if the build completes.

**Reply by akulbe** [reply]
This is the part that I don't understand. The network is flat. A single subnet. So I'm not sure why it would be unreachable.

**Reply by tenthirtyam** [reply]
Assuming that you're Linux, have you configured IPTables?

**Reply by akulbe** [reply]
🤦🏻‍♂️ it helps when you reload the firewall. 

Sorry for this. I figured it was something I missed. 

On another note… THANK YOU for this work. 🙌🏻

**Reply by tenthirtyam** [reply]
That's great to hear - you're welcome!

Be sure to give us a GitHub ⭐!

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.
